design-editor/src/system/libraries/js-library.js

```diff
@@ -32,44 +31,16 @@ class JSLibrary extends Library {
 	createHTMLElement(currentFile) {
 		this.element = this.element || document.createElement('script');
 		this.setAttribute('type', this.type);
-		if (this._fileName) {
+		if (this.fileName) {
 			// TODO: uncoment this line when fix
 			// this.setAttribute('src', this.getRelativePath(currentFile));
 			this.setAttribute('src', this.getAbsolutePath(true));
-			this.setAttribute(this.createDataAttribute(), '');
 		}
 		return this.element;
 	}
 
-	/**
-	 * Inserts content into library
-	 * both as internal script or copy proper file
-	 * @param  {function} callback callback function after insertion.
-	 */
-	insertLibContent(callback) {
-		if (this._fileName) {
-			this.copyLibFile(callback);
-		} else {
-			this.element.textContent = this.content;
-		}
-	}
-
-	/**
-	 * Set element attribute
-	 * @param {string} key attribute key
-	 * @param {string} value attribute value
-	 */
-	setAttribute(key, value='') {
-		this.addAttribute(key,value);
-		this.element.setAttribute(key, value);
-	}
-
-	createDataAttribute() {
-		return `data-${basename(this._fileName, extname(this._fileName))}`;
-	}
-
 	getSelector() {
```

Comment:
can we move getSelector to the Library class?

Reply:
For now, I leave it as is. It's a good point to refactor in the next iteration